config/nrfconnect/chip-module/Kconfig

```diff
@@ -202,6 +202,54 @@ config CHIP_FACTORY_RESET_ERASE_NVS
 	bool
 	default y if CHIP_FACTORY_DATA || CHIP_FACTORY_DATA_CUSTOM_BACKEND
 
+config CHIP_WIFI_CONNECTION_RECOVERY_ENABLE
+	bool "Enable WiFi connection recovery feature"
+	depends on CHIP_WIFI
+	help
+	  Enables the connection recovery WiFi feature which allows the device 
+	  performing the next scanning and connection to the known network, 
+	  when the connection has been lost. 
+	  The re-scan time starts from a minimum interval specified by 
+	  CHIP_WIFI_CONNECTION_RECOVERY_MINIMUM_INTERVAL, and with every connection attempt 
+	  the wait time is doubled up to CHIP_WIFI_CONNECTION_RECOVERY_MAXIMUM_INTERVAL. 
+
+config CHIP_WIFI_CONNECTION_RECOVERY_MINIMUM_INTERVAL
+	int "Define the minimum connection recovery time interval in milliseconds"
+	depends on CHIP_WIFI_CONNECTION_RECOVERY_ENABLE
+	default 500
+	help
+	  Specifies the maximum connection recovery interval (in milliseconds).
+
+config CHIP_WIFI_CONNECTION_RECOVERY_MAXIMUM_INTERVAL
+	int "Define the maximum connection recovery time interval in milliseconds"
+	depends on CHIP_WIFI_CONNECTION_RECOVERY_ENABLE
+	default 3600000 # 1 hour
+	help
+	  Specifies the maximum connection recovery interval (in milliseconds).
+
+config CHIP_WIFI_CONNECTION_RECOVERY_JITTER_INTERVAL
+	int "Define the connection recovery jitter in milliseconds"
+	depends on CHIP_WIFI_CONNECTION_RECOVERY_ENABLE
+	default 2000
+	help
+	  Specifies the maximum connection recovery jitter interval (in milliseconds).
+	  Once the wait time reaches the maximum value a random jitter interval is
+	  added added to it to avoid periodicity. The random jitter is selected
+	  within range [-JITTER; +JITTER]. The random jitter value is added only
```

Comment:
I don't get it. If our JITTER is e.g. 2000, it means the range is -2000;+2000, right? And as a result we are adding or subtracting some value from the MAX interval. I don't see any subtracting in the implementation below. Btw. if my description is right and that's what you wanted to achieve we should add assert checking if JITTER < MAX to prevent reaching negative values after subtracting.

Reply:
I made a mistake here, I will fix it.